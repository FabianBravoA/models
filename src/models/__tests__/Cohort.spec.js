const Campus = require('../Campus');
const Cohort = require('../Cohort');


describe('Cohort', () => {
  it('should fail validation when campus not an ObjectId', () => {
    const cohort = new Cohort({
      campus: 'lim',
      program: 'bc',
      track: 'core',
      generation: 9,
      start: new Date(),
      end: new Date(),
    });
    return cohort.validate()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should fail validation when bootcamp related cohort and no generation', () => {
    const campus = new Campus({});
    const cohort = new Cohort({
      campus: campus._id,
      program: 'bc',
      track: 'core',
      start: new Date(),
      end: new Date(),
    });
    return cohort.validate()
      .catch(err => expect(err.message).toBe('Generation is required for program type bc'));
  });

  it('should validate campus when all good', () => {
    const campus = new Campus({});
    const cohort = new Cohort({
      campus: campus._id,
      program: 'bc',
      track: 'core',
      generation: 9,
      start: new Date(),
      end: new Date(),
    });
    return cohort.validate();
  });
});
