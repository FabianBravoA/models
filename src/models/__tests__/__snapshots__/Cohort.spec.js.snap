// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should fail validation when campus not an ObjectId 1`] = `
Object {
  "campus": [CastError: Cast to ObjectID failed for value "lim" at path "campus"],
}
`;
